core/src/main/java/com/linecorp/armeria/common/logging/DefaultRequestLog.java

```diff
@@ -428,7 +428,8 @@ private void updateFlags(int flags) {
     private static void completeSatisfiedFutures(RequestLogFuture[] satisfiedFutures, RequestLog log,
                                                  RequestContext ctx) {
         if (!ctx.eventLoop().inEventLoop()) {
-            ctx.eventLoop().execute(() -> completeSatisfiedFutures(satisfiedFutures, log, ctx));
+            ctx.eventLoop().withoutContext().execute(
+                    () -> completeSatisfiedFutures(satisfiedFutures, log, ctx));
```

Comment:
Q) Why was the context removed? Shouldn't the request context be propagated to the callbacks of `satisfiedFutures`?

Reply:
It seemed like we were making an effort to clear the request context when completing logs.

e.g.
https://github.com/line/armeria/blob/72ebfe12abf7804723a490fe7acfe3a45d46089f/core/src/main/java/com/linecorp/armeria/client/HttpClientDelegate.java#L242-L247

I saw `ContextCancellationTest` failing occasionally due to this, so I modified to use `withoutContext`.

Reply:
It was an inevitable choice Netty can handle pending messages in batch mode that can cause a request context leak.

If an action does not perform sequentially in the callbacks of Netty operations, I think a context-aware executor is generally preferred. 

I am not strong here. If you have any other workarounds, it is okay to keep it.

Reply:
> Netty can handle pending messages in batch mode that can cause a request context leak

I see, I think I misunderstood the intention then. My original guess was that if request logs for multiple requests are waited on a single condition, users would easily see a request context leak.

I'm not sure what you mean by batch mode though. Channels are tied to a single thread, and within a single thread as long as the `try...close` syntax is used context leaks weren't possible.

Reply:
> try...close syntax is used context leaks weren't possible.

Context leaks could happen even with `try...close`. 
`if request logs for multiple requests are waited on a single condition` seems to mean the same thing, but I'll explain it just in case.

Say we have a simple service that adds a callback to `response.whenComplete()`.
```java
try (ctx.push()) {
   val response = HttpResponse.streaming();
   response.write(...);
   response.whenComplete().thenAccept(unused -> {
      // Do something else.
   })
}
```

- The code pushes `ctx` temporary using `try with resource` block.
- `response.whenComplete()` may be completed when `WriteFutureListener.operationComplete()` is invoked if it is a unary type.
https://github.com/line/armeria/blob/cd1973b35fc0326bf7369e1a351d74a1434b911f/core/src/main/java/com/linecorp/armeria/server/AggregatedHttpResponseHandler.java#L92
- Netty `Channel` stores write requests to `ChannelOutboundBuffer`.
- `response.write()` for `request_1` to `request_9` are added to `ChannelOutboundBuffer` and.
- When `response.write()` for `request_10` is written, the channel can flush the outbound data so `request_1` to `request_10` are flushed in the same call (task).
- Without `try (SafeCloseable ignored = RequestContextUtil.pop())`, the callbacks of `request_1-9.whenComplete()` will see the context of `request_10`.

It would be impossible to leak the context if `completeSatisfiedFutures()` is executed asynchronously. So I thought it was safe to remove `.withoutContext()`. 

However, as you said, contexts are not propagated in some cases. I agree with keeping the current style in terms of consistency.

Reply:
Thanks for the explanation 👍 

> When response.write() for request_10 is written, Channel can flush the outbound data so request_1 to request_10 are flushed in the same call (task).

I understood that even if `request_1` to `request_10` is flushed together in a single task, they will be executed sequentially.
e.g.
```java
ClientRequestContext ctx1 = null;
ClientRequestContext ctx2 = null;
// represents a single task
CommonPools.workerGroup().execute(() -> {
    // request 1
    try (SafeCloseable sf = ctx1.push()) {
        ctx1.logBuilder().endResponse();   
    }
    // request 2
    try (SafeCloseable sf = ctx2.push()) {
        ctx2.logBuilder().endResponse();
    }
});
```

>  I agree with keeping the current style in terms of consistency.

👍  for me as well. Ultimately, my thought was that it depends more on how users are using the API - for this reason it is probably more important that the context is pushed (or not) consistently.

e.g. I can imagine the following scenario
```java
HttpResponseWriter res1 = null;
HttpResponseWriter res2 = null;
CompletableFuture.allOf(ctx1.log().whenComplete(), ctx2.log().whenComplete())
                 .handle((v, t) -> {
                     res1.close();
                     res2.close();
                     // when the res is completed, the same thread may call ctx.push somewhere
                     return null;
                 });
```

Reply:
I didn't expect this change to be so controversial 😅  Let me just revert this change since this isn't really critical to this PR